### PR TITLE
`filt` perf optimizations

### PR DESCRIPTION
**Pull request by wheeheee**
- instead of allocating for every column, reuse `si` (different from input `si`)
- `filt` currently errors for `N>2` (edit: if the trailing dimensions aren't all of length 1); this PR doesn't change that

### CODE REVIEW

**Comment by ViralBShah** [comment]
@wheeheee The DSP.jl package needs new maintainers. Do you feel comfortable with taking on that role?

**Comment by ViralBShah** [comment]
@martinholters Ok to merge? Hope you don't mind the ping.

**Review by martinholters — COMMENTED** [review]
> @martinholters Ok to merge? Hope you don't mind the ping.

Don't mind at all. In principle ok with me. I'm not too excited about all those `muladd`s; I find them less readable than the straight `a*b+c` and don't think they bring benefits to justify that everywhere. That said, now that all the replacements have been done, I don't think it's worth it to go through them and see whether they bring any tangible benefits.

One thing I'm not sure about and would like a word of explanation is all those deleted `filt(self, ...)` methods. What's up with those? Why don't we need them anymore?

**Comment by martinholters** [comment]
Ah, yes, of course. I had missed the edited one.

**Review by martinholters — APPROVED** [review]

**Comment by ViralBShah** [comment]
Merge? Also @wheeheee would you be ok to be invited to the organization to help maintain these packages?

**Comment by ViralBShah** [comment]
@wheeheee Thanks for helping with the maintenance here. Since we have approval here from @martinholters, it would be great to rebase and get this one merged.

**Comment by martinholters** [comment]
Thanks @wheeheee, any help here is appreciated. And also thanks @ViralBShah for pushing this forward.

**Comment by ViralBShah** [comment]
Thanks @martinholters for all the work on this package, and @wheeheee for jumping in. We should see if there are other contributors who would like to step up as well. I'm just trying to help out - amazing how the small improvements over the last few days are making a big cosmetic difference at least.

**Pull request merged**
